### ISSUE TITLE
makeArray does not work with an object

### ISSUE THREAD

**Issue by phillipskevin** [canjs/can-util#209]
With `can.makeArray` in canjs 2.3:
```
can.makeArray({foo: 'bar', abc: 'xyz'}); // [{ abc: "xyz", foo: "bar" }]
```

with `can-util/js/make-array/make-array`:
```
can.makeArray({foo: 'bar', abc: 'xyz'}); // []
```

Related to https://github.com/canjs/canjs/issues/3116.

**Comment by justinbmeyer** [comment]
We were talking about this today in training actually.  I'm not sure what the behavior should be .  What should makeArray( null ) do for example?

My instinct is to throw if someone pass something that can't be converted to an array.  Someone is doing something wrong and we should let them know about it.

**Comment by phillipskevin** [comment]
I'm not sure what `makeArray(null)` should do, maybe it should return `[null]` like `new Array(null)` does.

I'm fine either supporting this or updating the migration guide to mention it no longer works the same way as 2.3.

**Comment by sarathnair** [comment]
What is expected here?

can.makeArray({foo: 'bar', abc: 'xyz'}); -> should this return [{ abc: "xyz", foo: "bar" }]?

And what about makeArray(null)?

**Comment by justinbmeyer** [comment]
What does jQuery do in these situations?  Is there a proposal for this sort of thing in browsers?

**Comment by sarathnair** [comment]
jQuery does the following:

$.makeArray({foo: 'bar', abc : 'xyz'}) -> // [{"foo":"bar","abc":"xyz"}]
$.makeArray(null) -> // [ ]

**Comment by phillipskevin** [comment]
Closed by #216.
